qt/widgets/common/inc/MantidQtWidgets/Common/QtPropertyBrowser/qteditorfactory.h

```diff
@@ -104,24 +104,113 @@ class QColor;
+*/
+template <class SpinBox>
+void QtSpinBoxFactoryBase<SpinBox>::connectPropertyManager(
+    QtIntPropertyManager *manager) {
+  connect(manager, SIGNAL(valueChanged(QtProperty *, int)), this,
+          SLOT(slotPropertyChanged(QtProperty *, int)));
+  connect(manager, SIGNAL(rangeChanged(QtProperty *, int, int)), this,
+          SLOT(slotRangeChanged(QtProperty *, int, int)));
+  connect(manager, SIGNAL(singleStepChanged(QtProperty *, int)), this,
+          SLOT(slotSingleStepChanged(QtProperty *, int)));
+}
+
+/**
+    \internal
+
+    Reimplemented from the QtAbstractEditorFactory class.
+*/
+template <class SpinBox>
+QWidget *QtSpinBoxFactoryBase<SpinBox>::createEditorForManager(
+    QtIntPropertyManager *manager, QtProperty *property, QWidget *parent) {
+  auto *editor = d_ptr->createEditor(property, parent);
+  editor->setSingleStep(manager->singleStep(property));
+  editor->setRange(manager->minimum(property), manager->maximum(property));
+  editor->setValue(manager->value(property));
+  editor->setKeyboardTracking(false);
+
+  connect(editor, SIGNAL(valueChanged(int)), this, SLOT(slotSetValue(int)));
+  connect(editor, SIGNAL(destroyed(QObject *)), this,
+          SLOT(slotEditorDestroyed(QObject *)));
+  return editor;
+}
+
+/**
+    \internal
+
+    Reimplemented from the QtAbstractEditorFactory class.
+*/
+template <class SpinBox>
+void QtSpinBoxFactoryBase<SpinBox>::disconnectPropertyManager(
+    QtIntPropertyManager *manager) {
+  disconnect(manager, SIGNAL(valueChanged(QtProperty *, int)), this,
+             SLOT(slotPropertyChanged(QtProperty *, int)));
+  disconnect(manager, SIGNAL(rangeChanged(QtProperty *, int, int)), this,
+             SLOT(slotRangeChanged(QtProperty *, int, int)));
+  disconnect(manager, SIGNAL(singleStepChanged(QtProperty *, int)), this,
+             SLOT(slotSingleStepChanged(QtProperty *, int)));
+}
+
+class QSpinBoxReadOnly : public QSpinBox {
+  Q_OBJECT
+public:
+  QSpinBoxReadOnly(QWidget *parent = nullptr) : QSpinBox(parent) {
+    this->lineEdit()->setReadOnly(true);
+  };
+};
+
+class EXPORT_OPT_MANTIDQT_COMMON QtSpinBoxFactoryReadOnly
+    : public QtSpinBoxFactoryBase<QSpinBoxReadOnly> {
+  Q_OBJECT
+public:
+  QtSpinBoxFactoryReadOnly(QObject *parent = nullptr);
+  ~QtSpinBoxFactoryReadOnly();
+
+private:
+  Q_DECLARE_PRIVATE(QtSpinBoxFactoryReadOnly)
+  Q_DISABLE_COPY(QtSpinBoxFactoryReadOnly)
+  Q_PRIVATE_SLOT(d_func(), void slotPropertyChanged(QtProperty *, int))
+  Q_PRIVATE_SLOT(d_func(), void slotRangeChanged(QtProperty *, int, int))
+  Q_PRIVATE_SLOT(d_func(), void slotSingleStepChanged(QtProperty *, int))
+  Q_PRIVATE_SLOT(d_func(), void slotSetValue(int))
+  Q_PRIVATE_SLOT(d_func(), void slotEditorDestroyed(QObject *))
+};
+
+class EXPORT_OPT_MANTIDQT_COMMON QtSpinBoxFactory
+    : public QtSpinBoxFactoryBase<QSpinBox> {
+  Q_OBJECT
+public:
+  QtSpinBoxFactory(QObject *parent = nullptr);
+  ~QtSpinBoxFactory() override;
```

Comment:
Why do we override the destructor here and not in QtSpinBoxFactoryReadOnly?

Reply:
I've added an override to the QtSpinBoxFactory to be consistent with what was being done before.